core/templates/components/skill-selector/skill-selector.directive.ts

```diff
@@ -37,23 +39,131 @@ angular.module('oppia').directive('selectSkill', [
         function(
             $scope, $uibModal, $rootScope) {
           var ctrl = this;
-          $scope.selectSkill = function(skillId) {
-            $scope.selectedSkillId = skillId;
-          };
           ctrl.$onInit = function() {
-            $scope.sortedSkillSummaries = $scope.getSortedSkillSummaries();
-            $scope.skillSummariesInitial = [];
-            $scope.skillSummariesFinal = [];
+            $scope.selectedSkill = null;
+            $scope.categorizedSkills = $scope.getCategorizedSkills();
+            $scope.checkIfEmpty = function(skills) {
+              return (skills.length === 0);
+            };
+            $scope.setSelectedSkillId = function() {
+              $scope.selectedSkillId = $scope.selectedSkill;
+            };
+            $scope.hasUntriagedSkills = function(topicName) {
+              return (topicName === 'untriaged_skills');
```

Comment:
Ditto, about topicName being equal to 'untriaged_skills'

Reply:
This function is used to distinguish the key  `untriaged_skills` from other keys in the `categorizedSkills` dict.  because the structure of the value of this key is differs from other keys(i.e topic names).
Pls refer to this [comment](https://github.com/oppia/oppia/pull/9280#discussion_r429610860)